app/views/handlers/submission.py

```diff
@@ -42,7 +42,10 @@ def submit_questionnaire(self):
         self._store_submitted_time_and_display_address_in_session()
         submission_schema: Mapping = self._schema.get_submission() or {}
 
-        if not submission_schema.get("view_submitted_response"):
+        if submission_schema.get("view_submitted_response"):
+            self._questionnaire_store.submitted_at = datetime.now(tz=tzutc())
```

Comment:
This shouldn't be generated here as it won't always match the `submitted_at` value in the payload. So need to either generate the datetime upfront and pass to convert answer and this or just reuse the value from the payload here.

Reply:
Does it need to? There is no requirement to find submitted by the payload. In fact this was purely being used for deletion after 24 hours so we would never be in a position to find it in time anyway

Reply:
However I will change it